### ISSUE TITLE
Promise fails on OSX Chrome 42

### ISSUE THREAD

**Issue by davidhund** [bramstein/fontfaceobserver#23]
For some reason I can't get FFO to work on my Chrome 42, OSX.

I've added FFO (standalone) just above `</body>` and initialize FFO as:

```
// main.js
(function( win, doc, undefined ){
  var observer = new FontFaceObserver('Open Sans', { weight: 300 } );
  observer.check()
    .then(function(result){
      console.log('Fonts loaded');
      doc.documentElement.className += " fonts-loaded";
    }, function(error){
      console.log('Fonts NOT loaded');
    });
}( this, document ));
```

This works in Fx, Safari and **even my Chrome Mobile 42 on my Nexus4**, but *not on my Desktop Chrome*…

The Promise does return a `z` `observer` object that looks ok to me:

```
{
  family: "Open Sans"
  featureSettings: "normal"
  stretch: "stretch"
  style: "normal"
  variant: "normal"
  weight: 300
}
```

I must be missing something but I can't see why Chrome fails.

## Some details:

- I am using local fonts created by https://google-webfonts-helper.herokuapp.com/fonts/open-sans?subsets=latin ("Best Support" font-face syntax)
- Running a local Browsersync live-reloading dev-server

Reading up on #1 I've considered the scroll-stuff to potentially conflict with Browsersync's scroll-syncing stuff, but that's just a wild guess.

Any thoughts?

**Comment by bramstein** [comment]
Thanks for reporting this. Unfortunately I haven't been able to reproduce this using the sample code below. Is the promise rejected on your machine, or is it incorrectly resolved?

```html
<!doctype html>
<html>
  <head>
    <style>
      @font-face {
        font-family: 'Open Sans';
        font-style: normal;
        font-weight: 400;
        src: url('fonts/open-sans-v13-latin-regular.eot'); /* IE9 Compat Modes */
        src: local('Open Sans'), local('OpenSans'),
             url('fonts/open-sans-v13-latin-regular.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
             url('fonts/open-sans-v13-latin-regular.woff2') format('woff2'), /* Super Modern Browsers */
             url('fonts/open-sans-v13-latin-regular.woff') format('woff'), /* Modern Browsers */
             url('fonts/open-sans-v13-latin-regular.ttf') format('truetype'), /* Safari, Android, iOS */
             url('fonts/open-sans-v13-latin-regular.svg#OpenSans') format('svg'); /* Legacy iOS */
      }
    </style>
  </head>
  <body>
    <script src="../fontfaceobserver.standalone.js"></script>
    <script>
      // main.js
      (function( win, doc, undefined ){
        var observer = new FontFaceObserver('Open Sans', { weight: 300 } );
        observer.check()
          .then(function(result){
            console.log('Fonts loaded');
            doc.documentElement.className += " fonts-loaded";
          }, function(error){
            console.log('Fonts NOT loaded');
          });
      }( this, document ));
    </script>
  </body>
</html>
```

**Comment by davidhund** [comment]
Thanks for the quick reply @bramstein — Your test works with me too. It seems to be in my (linked) stylesheet. Investigating…

**Comment by davidhund** [comment]
@bramstein I think I've found the issue!
I am using https://github.com/jonathantneal/sanitize.css
Sanitize adds `* { font-size: inherit; }` before my styles:

```
/* styles.css
---------------- */
@font-face { ... }

/* Added by Sanitize.css */
* { font-size: inherit; ... }

/* Rest of my styles */
:root { font-family: Helvetica; }

/* Open Sans + default fallback */
.fonts-loaded { font-family: 'Open Sans'; }
```
.. removing the `* { font-size: inherit; ... }` fixes it in chrome.
I guess it messes with *your* hidden `div` measurements…

Any ideas on how to work around this issue?

**Comment by bramstein** [comment]
@davidhund Good find! I think you're right that it probably overrides something in my test divs. I think adding `!important` should fix that.

**Comment by davidhund** [comment]
!important all the things! :boom:

**Comment by bramstein** [comment]
@davidhund I added `* { font-size: inherit; }` to my test page, and it still works. Is there anything else in that selector?

**Comment by davidhund** [comment]
@bramstein It's difficult to pinpoint (maybe because of cache?).
I could provide you with my test files if you wish.

One other rule I've noticed is a `text-shadow` on the body:

```
/* styles.css
---------------- */
@font-face { ... }

/* Added by Sanitize.css */
* { font-size: inherit; ... }

/* Rest of my styles */
:root { font-family: Helvetica; }

/* Open Sans + default fallback */
.fonts-loaded { font-family: 'Open Sans'; }

/* When this is active _and_ font-size: inherit is, FFO fails me... */
body { text-shadow:0 .1em 0 rgba(0,0,0,0.08); }
```

The **combination** of the `* { font-size: inherit; } ` _and_ `body { text-shadow: ...; }` makes FFO fail, it seems.

Could you try adding my `text-shadow` (on `body`?) at the end of your test CSS?

**Comment by davidhund** [comment]
I've made two limited test-cases on codepen.io. The *only* difference is that in the latter (`NqdXPv`) I removed the `* { font-size: inherit; }`

Please test these in an INCOGNITO window.
I've seen a lot of strange cache(?) issues…

- Failing (*with* `* { font-size: inherit; }`): http://codepen.io/davidhund/full/MwJOMV/ (red)
- Success (*without* `* { font-size: inherit; }`): http://codepen.io/davidhund/full/NqdXPv/ (green)

**Comment by bramstein** [comment]
Yup, thanks a lot. That's reproducible. I'll dig into it one of these days. Thanks, these test cases really help.

**Comment by davidhund** [comment]
Phew! In Chrome's normal mode I've had intermittent successes, so it could be some caching/network issues. 

Also: I still wonder if the `text-shadow` on `body` has some effect.

I've asked people on twitter to test too, let me know if there's anything else I can do…

**Comment by bramstein** [comment]
Sorry for the late response, some things came up that prevented me from working on this. I fixed the issue in the latest release. It turns out `* { font-size: inherit; }` was indeed the culprit, it changed how the font load detection worked, which in this case made it fail. I've added explicit font sizes to all test elements, so the issue no longer happens.

Thanks again for the test cases, they were really helpful in tracking down this issue.
